### PR TITLE
Add 'OpenSceneGraph' package

### PR DESCRIPTION
**Pull request by t0p4**
* I've followed [this guide](https://docs.hunter.sh/en/latest/creating-new/create/cmake.html)
  step by step carefully. **[Yes]**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/t0p4/hunter-7qcko/builds/21173038
  * https://travis-ci.org/t0p4/hunter/builds/470960340>

* I have submitted CI configs to https://github.com/ingenue/hunter targeting `pkg.template` branch,
  see this merged pull request https://github.com/ingenue/hunter/pull/320

---

### CODE REVIEW

File: cmake/projects/OpenSceneGraph/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    OpenSceneGraph
+    VERSION
+    3.6.3-p0
+    URL
+    "https://github.com/t0p4/OpenSceneGraph/archive/v3.6.3-p0.tar.gz"
```
Comment (ruslo) [inline cmake/projects/OpenSceneGraph/hunter.cmake]: We have talked about it at least 6 times:
* https://github.com/ruslo/hunter/pull/1670#discussion_r241994878
* https://github.com/ruslo/hunter/pull/1631#discussion_r236286364
* https://github.com/ruslo/hunter/pull/1626#discussion_r235530554
* https://github.com/ruslo/hunter/pull/1617#discussion_r232957871
* https://github.com/ruslo/hunter/pull/1596#discussion_r228043064
* https://github.com/ruslo/hunter/pull/1574#discussion_r223204344

Really?

File: docs/packages/pkg/OpenSceneGraph.rst [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+.. _pkg.OpenSceneGraph:
+
+OpenSceneGraph
+==============
+
+-  `Official <https://github.com/openscenegraph/OpenSceneGraph>`__
+-  `Hunterized <https://github.com/hunter-packages/OpenSceneGraph>`__
+-  `Example <https://github.com/ruslo/hunter/blob/master/examples/OpenSceneGraph/CMakeLists.txt>`__
+-  Added by `t0p4 <https://github.com/t0p4>`__ (`pr-N <https://github.com/ruslo/hunter/pull/N>`__)
```
Comment (ruslo) [inline docs/packages/pkg/OpenSceneGraph.rst]: https://docs.hunter.sh/en/latest/creating-new/create/cmake.html#pull-requests

File: examples/OpenSceneGraph/boo.cpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#include <osg/CoordinateSystemNode>
+#include <osg/Switch>
+#include <osg/Types>
+#include <osgText/Text>
+
+
+int main() {
+}
+
+int main() {
```
Comment (ruslo) [inline examples/OpenSceneGraph/boo.cpp]: There are two `main` function, please cleanup.

File: cmake/projects/OpenSceneGraph/hunter.cmake [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+hunter_add_version(
+    PACKAGE_NAME
+    OpenSceneGraph
+    VERSION
+    "3.6.3-p0"
+    URL
+    "https://github.com/hunter-packages/OpenSceneGraph/archive/v3.6.3-p0.tar.gz"
+    SHA1
+    ed5e52f56fe34497342a88fa894350336e4f8033
```
Comment (ruslo) [inline cmake/projects/OpenSceneGraph/hunter.cmake]: SHA1 is not correct

**Comment by t0p4** [comment]
Could you execute test again

**Comment by ruslo** [comment]
> macOS build failed:

Linux fail as well:
```
CMake Error at cmake_install.cmake:113 (file):
  file cannot create directory: /cmake/OpenSceneGraph.  Maybe need
  administrative privileges.
```

Looks like package trying to install to root `/` folder instead of local folder.

As a summary nothing is working. Have you test it on your local machine with `TOOLCHAIN=clang-libcxx14-fpic PROJECT_DIR=examples/OpenSceneGraph ./jenkins.py`?

**Comment by t0p4** [comment]
Yes, I've tested. Was succeeded with this toolchain and clang-fpic

**Comment by t0p4** [comment]
The package was successfully installed before the hunter and after the changes. It's the last package in my project without him I'm in freeze

**Comment by ruslo** [comment]
> Linux fail as well:

Still failed, same error. You can reproduce the issue using docker, see:
* https://github.com/hunter-packages/travis-linux-docker

**Comment by t0p4** [comment]
> > macOS build failed:
> 
> Linux fail as well:
> 
> ```
> CMake Error at cmake_install.cmake:113 (file):
>   file cannot create directory: /cmake/OpenSceneGraph.  Maybe need
>   administrative privileges.
> ```
> Looks like package trying to install to root `/` folder instead of local folder.
> 
> As a summary nothing is working. Have you test it on your local machine with `TOOLCHAIN=clang-libcxx14-fpic PROJECT_DIR=examples/OpenSceneGraph ./jenkins.py`?

I've foud it. fixed

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.23.83

**Pull request merged**